ipatests/test_ipalib/test_config.py

```diff
@@ -563,7 +563,7 @@ def test_finalize_core(self):
         # Test using DEFAULT_CONFIG:
         defaults = dict(constants.DEFAULT_CONFIG)
         (o, home) = self.finalize_core(None, **defaults)
-        assert list(o) == sorted(defaults)
+        assert set(o).issuperset(set(defaults))
```

Comment:
I'm not sure how you'd like to make sure `o` doesn't contain `fips_mode`. `Env` object does not allow deletion and the only way I can think of would be to violate the object's "private" properties.

I propose we either:

- always set fips_mode: to `True`, `False` or `None`
- accept that `Env` can now have optional attributes and use the test proposed above